opf/openproject
# Add LDAP env seeder

https://community.openproject.org/work_packages/46216

**How to tests**

- Add file `ldap.env`

```
export OPENPROJECT_SEED_LDAP_FOO_HOST="localhost"
export OPENPROJECT_SEED_LDAP_FOO_PORT="12389"
export OPENPROJECT_SEED_LDAP_FOO_SECURITY="plain_ldap" # values of the enum
export OPENPROJECT_SEED_LDAP_FOO_TLSVERIFY="false"
export OPENPROJECT_SEED_LDAP_FOO_BINDUSER="uid=admin,ou=system"
export OPENPROJECT_SEED_LDAP_FOO_BINDPASSWORD="secret"
export OPENPROJECT_SEED_LDAP_FOO_BASEDN="dc=example,dc=com"
export OPENPROJECT_SEED_LDAP_FOO_FILTER="(uid=*)"
export OPENPROJECT_SEED_LDAP_FOO_SYNC__USERS="true"
export OPENPROJECT_SEED_LDAP_FOO_LOGIN__MAPPING="uid"
export OPENPROJECT_SEED_LDAP_FOO_FIRSTNAME__MAPPING="givenName"
export OPENPROJECT_SEED_LDAP_FOO_LASTNAME__MAPPING="sn"
export OPENPROJECT_SEED_LDAP_FOO_MAIL__MAPPING="mail"
export OPENPROJECT_SEED_LDAP_FOO_ADMIN__MAPPING=""
export OPENPROJECT_SEED_LDAP_FOO_GROUPFILTER_BAR_BASE="ou=groups,dc=example,dc=com"
export OPENPROJECT_SEED_LDAP_FOO_GROUPFILTER_BAR_FILTER="(cn=*)"
export OPENPROJECT_SEED_LDAP_FOO_GROUPFILTER_BAR_SYNC__USERS="true"
export OPENPROJECT_SEED_LDAP_FOO_GROUPFILTER_BAR_GROUP__ATTRIBUTE="dn"
```

- run `bundle exec rake ldap_groups:development:ldap_server` in a second console window

- (Precondition for filter: Have active EE)

- run `source ldap.env ; bundle exec rake db:seed`

- start server, check if connection + filter + groups are there
